cli/shared/src/main/scala/com/lightbend/rp/reactivecli/http/Http.scala

```diff
@@ -44,12 +46,22 @@ object Http {
 
           val location = response.headers("Location")
 
-          if (visitedUrls.contains(location) || visitedUrls.length >= settings.maxRedirects)
+          if (visitedUrls.contains(location) || visitedUrls.length >= settings.maxRedirects) {
+            logger.debug("No more redirects allowed")
             Future.failed(InfiniteRedirect(visitedUrls))
-          else
-            doRequest(
-              HttpRequest(location).copy(requestFollowRedirects = request.requestFollowRedirects),
-              location :: visitedUrls)
+          } else {
+            println(location)
+            if (location.indexOf("X-Amz-Credential") > -1) {
+              // S3 based registry doesn't accept authorization header on redirect
+              logger.debug("Performing S3 redirect")
+              doRequest(
+                HttpRequest(location, tlsValidationEnabled = Some(true), requestFollowRedirects = request.requestFollowRedirects),
+                location :: visitedUrls)
+            } else {
+              doRequest(
+                request.copy(location), location :: visitedUrls)
```

Comment:
Should this fall back to what the code is currently doing?

```
            doRequest(
              HttpRequest(location).copy(requestFollowRedirects = request.requestFollowRedirects),
              location :: visitedUrls)
 ```

Reply:
It shouldn't because the code

`doRequest(
              HttpRequest(location).copy(requestFollowRedirects = request.requestFollowRedirects),
              location :: visitedUrls)`

doesn't copy auth credentials which is required by a docker registry.